kuka_rsi_hw_interface/src/kuka_hardware_interface.cpp

```diff
@@ -152,16 +152,66 @@ CallbackReturn KukaRSIHardwareInterface::on_activate(const rclcpp_lifecycle::Sta
     bytes = server_->recv(in_buffer_);
   }
 
-  rsi_state_ = RSIState(in_buffer_);
+  if (!command_handler_.Decode(in_buffer_, UDP_BUFFER_SIZE)) {
+    RCLCPP_ERROR(rclcpp::get_logger("KukaRSIHardwareInterface"), "Decode failed");
+    return CallbackReturn::FAILURE;
+  }
+
+  // Position data
+  hw_states_[0] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A1") *
+    KukaRSIHardwareInterface::D2R;
+  hw_states_[1] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A2") *
+    KukaRSIHardwareInterface::D2R;
+  hw_states_[2] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A3") *
+    KukaRSIHardwareInterface::D2R;
+  hw_states_[3] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A4") *
+    KukaRSIHardwareInterface::D2R;
+  hw_states_[4] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A5") *
+    KukaRSIHardwareInterface::D2R;
+  hw_states_[5] = command_handler_.GetState().GetElement("AIPos")->GetParam<double>("A6") *
+    KukaRSIHardwareInterface::D2R;
```

Comment:
You could consider using the [angles](https://index.ros.org/p/angles/) package for these kinds of conversions ([angles::from_degrees(..)](https://docs.ros.org/en/api/angles/html/namespaceangles.html#ac427c1e73e487dcfca841672f719990a), etc).

Reply:
thanks for the tip, I corrected it